clients/common/clientComm.py

```diff
@@ -37,6 +37,58 @@ def stopComm(self):
         self.runComm = False
         self.wsConnected = False
 
+
+    def sendReceiveMsg(self, req, timeout): 
+        jsonDump = json.dumps(req)
+        self.sendMsgQueue.put(jsonDump)
+        while True:
+            try:
+                resp = self.recvMsgQueue.get(timeout = timeout)
+                respJson =  json.loads(resp) 
+                if str(req["requestId"]) == str(respJson["requestId"]):
```

Comment:
I think this loop is clever, but does not help you, when all helper functions are using static ids (i.e. you may notice the difference between a set and get, but not between different sets

I suggest a globally counting id for all helpers (maybe safest), or a random number (shoudl besafe enough if large enough range) , or separately counting ones for each message type with diffeerent prefixes (not the most clever I think, because they might run into each other)

Reply:
you are right, but I think, it should be fixed in another PR so that the PR does not become more complicated. because the behavior currently is better than master, not worse. An there is some other things, that I also like to improve from test client, but not in this PR.